library/std/src/net/parser.rs

```diff
@@ -140,10 +148,7 @@ impl<'a> Parser<'a> {
                 *slot = p.read_separator('.', i, |p| {
                     // Disallow octal number in IP string.
                     // https://tools.ietf.org/html/rfc6943#section-3.1.1
-                    match (p.peek_char(), p.read_number(10, None)) {
-                        (Some('0'), Some(number)) if number != 0 => None,
-                        (_, number) => number,
-                    }
+                    p.read_number(10, None, false)
```

Comment:
Just an advice. 

```suggestion
                    p.read_number(10, Some(3), false)
```

With this, we can avoid `if s.len() > 15 {` check in `Ipv4Addr::from_str`.

Reply:
The `s.len() >= 15` check is just an optimization to avoid parsing overly long IPs that can't possibly be valid, it's never been needed.